### PR TITLE
fix stateflow

### PR DESCRIPTION
**Pull request by arnaudgiuliani**
Hey Marcin,

here is the changes to help a `stateFlow` buffer `setState`, and be sure that we protect each `setState`

### CODE REVIEW

**Comment by arnaudgiuliani** [comment]
Concept of ActorFlow has been merged into the main DataFlow

File: uniflow-core/src/main/kotlin/io/uniflow/core/flow/DataFlow.kt [hunk 1]
```diff
@@ -131,15 +119,25 @@ interface DataFlow {
...
+//    }
     fun onAction(action: StateAction) {
         coroutineScope.apply {
             if (isActive) {
-                UniFlowLogger.log("DataFlow.onAction run $action")
-                launchOnIO {
-                    proceedAction(action)
-                }
+                UniFlowLogger.log("DataFlow.onAction offer action $action")
+                actorFlow.offer(action)
```
Comment (MarcinChrapowiczCurve) [inline uniflow-core/src/main/kotlin/io/uniflow/core/flow/DataFlow.kt]: 👏

File: uniflow-test/src/test/kotlin/io/uniflow/test/data/TodoStackFlow.kt [hunk 2]
```diff
@@ -1,124 +1,127 @@
...
+//package io.uniflow.test.data
+//
+//import io.uniflow.core.flow.UIEvent
+//import io.uniflow.core.flow.UIState
+//import io.uniflow.core.flow.stateFlowFrom
+//import io.uniflow.core.sample.StackFlow
+//import io.uniflow.core.threading.onIO
+//import kotlinx.coroutines.delay
+//
+//class TodoStackFlow(private val repository: TodoRepository) : StackFlow() {
```
Comment (MarcinChrapowiczCurve) [inline uniflow-test/src/test/kotlin/io/uniflow/test/data/TodoStackFlow.kt]: Do you need this ?

File: uniflow-core/src/main/kotlin/io/uniflow/core/flow/DataFlow.kt [hunk 3]
```diff
@@ -131,15 +119,25 @@ interface DataFlow {
      * Execute the action & catch any flowError
      * @param action
      */
+//    fun onAction(action: StateAction) {
```
Comment (MarcinChrapowiczCurve) [inline uniflow-core/src/main/kotlin/io/uniflow/core/flow/DataFlow.kt]: Why we still keeping this ?

File: uniflow-test/src/test/kotlin/io/uniflow/test/data/TodoStackFlow.kt (hunk 2 above)
Comment (arnaudgiuliani) [reply uniflow-test/src/test/kotlin/io/uniflow/test/data/TodoStackFlow.kt]: later perhaps ... to clean tests :)

File: uniflow-core/src/main/kotlin/io/uniflow/core/flow/DataFlow.kt (hunk 3 above)
Comment (arnaudgiuliani) [reply uniflow-core/src/main/kotlin/io/uniflow/core/flow/DataFlow.kt]: It's the old `onAction`, that was working without Actor under the hood

**Pull request merged**